### ISSUE TITLE
Show accepted login names

### ISSUE THREAD

**Issue by mffap** [zitadel/zitadel#7411]
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [ ] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

As mentioned in https://github.com/zitadel/zitadel/issues/7409 users can already login with their username, email, or phone number, given the options are not disabled. Yet the login screen will always show "username@domain" as hint. This hint can be manually overwritten as workaround.

### Describe your ideal solution

As a user, I can see the allowed login methods on the login screen, so that I know what identifiers I can enter

### Environment

ZITADEL Cloud

### Additional Context

See login hints in popular apps like Facebook, Uber etc.

**Comment by mffap** [comment]
Workaround: Change login texts

https://<custom_domain>/ui/console/instance?id=logintexts
loginNamePlaceholder: "Login with email or phone number"
loginNameLabel: "Login to <Your Company Name>"

**Comment by doncicuto** [comment]
Hi @mffap and @hifabienne  I propose the following messages assuming that the code detects if either email, phone or both are disabled. Please check if these options are what you desire. I'll prepare the translations accordingly.

For instance login where all methods are enabled

![Captura desde 2024-02-20 19-41-55](https://github.com/zitadel/zitadel/assets/30386061/74f129a8-6ff5-4795-a5fc-755e699ab77e)

For instance login where email is disabled

![Captura desde 2024-02-20 19-42-48](https://github.com/zitadel/zitadel/assets/30386061/4923e575-d390-42f2-86bd-3388c66ebbc0)

For instance login where phone is disabled

![Captura desde 2024-02-20 19-43-11](https://github.com/zitadel/zitadel/assets/30386061/9909dd07-dfa3-403c-99ac-69ec98baece9)

For instance login where both phone and email are disabled

![Captura desde 2024-02-20 19-51-51](https://github.com/zitadel/zitadel/assets/30386061/d6f92226-f29e-49de-af63-88e84ddc642f)

@mffap while you suggest "Login with email or phone number" I'd like to know if you want to keep the username@domain in the message as phone and email can be both disabled

**Comment by mffap** [comment]
@doncicuto that's right, showing all options would be the most helpful to understand what's possible. 

Would it be possible to show the actual domain? I realize this gets a bit tricky, since the domain part contains also the sub-domain of the custom domain. I guess it would be actually `username@organization.custom-domain.zitadel.cloud` (where custom-domain is the actual domain of the instance) for the instance login. For an organization that has a verified domain, we should probably show `username@primary-domain` (where primary-domain is the actual primiary domain of the organization). 

Writing this, it seems a lot. My guess is that not many will use the generated domains, but use email/phone to let users login anyways. So there is an argument for showing this info only based on an option. Not sure what the best way is. Maybe @hifabienne has some thoughts on this. 

On another note, but very much related to this issue. If login with phone number is allowed, it should also be shown as "Login methods" on the user profile.

<img width="361" alt="Screenshot 2024-02-21 at 09 03 31" src="https://github.com/zitadel/zitadel/assets/1366906/fc78402d-a68f-4966-9d4e-5bcb635d383a">

@hifabienne any reason why we don't show the phone numner on "Login methods"?

**Comment by hifabienne** [comment]
I think if we take into account the settings, we should also check if the username does need the domain or not.
So in case of domain suffix enabled: username@domain
in case of suffix disabled username

The whole text is pretty long, also if we really take the domains the could get to long, to be able to show the rest (email, phone)
@peintnermax do you have any input from a ui perspective?

I think phone number just came later on so we probably didn't include that in the list.

**Comment by Vishwaspatel2401** [comment]
Hi! I'd like to work on this issue.

My approach would be to detect which login methods are enabled (username, email, phone) based on the instance settings, and dynamically update the login placeholder/label text to reflect the allowed options — similar to what @doncicuto proposed above.

Could a maintainer confirm this is still open and the approach makes sense before I start?
